### PR TITLE
first pass at adding build/packaging details for Python functions

### PR DESCRIPTION
**Pull request by jazwiecki**
*Issue #, if available:* #2478

*Description of changes:* Adds barebones documentation of build & package steps for Python functions, specifically around packaging dependencies.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cli/function/build-options.md
Comment (renebrandel) [inline docs/cli/function/build-options.md:6]: I'd move the Python section below the NodeJS section.

File: docs/cli/function/build-options.md
Comment (renebrandel) [inline docs/cli/function/build-options.md:8]: Remove the bold lettering here and move it and collapse it into the next paragraph as the first sentence.

File: docs/cli/function/build-options.md
Comment (renebrandel) [inline docs/cli/function/build-options.md:10]: Move "Amplify will run..." into its own paragraph.

Comment (renebrandel) [reply docs/cli/function/build-options.md:10]: Remove "comma" from "...directory during builds, using either..."

File: docs/cli/function/build-options.md
Comment (renebrandel) [inline docs/cli/function/build-options.md:12]: Shorten for readability. Remove "This means..." from both sentences. "The contents of..." and "Packages installed as..."

Comment (renebrandel) [reply docs/cli/function/build-options.md:12]: Shorten for clarity: Remove "properly". "... will not be packaged" is also accurate and shortened.

Comment (renebrandel) [inline docs/cli/function/build-options.md:12]: Shorten for readability: "as their only representation in the `site-packages` directory is an `.egg-link` file pointing to the local, editable code of the dependency" --> "as they are represented as an `.egg-link` file pointing to the local, editable code of the dependency."

**Review by renebrandel — CHANGES_REQUESTED** [review]

**Review by renebrandel — APPROVED** [review]

**Comment by mauerbac** [comment]
@jazwiecki , can you fix this merge conflict?

**Comment by jazwiecki** [comment]
@mauerbac yep, think I got it.

**Comment by mauerbac** [comment]
can we get a review @aws-amplify/amplify-cli

**Comment by jazwiecki** [comment]
AFAIK this is just waiting on a review from the CLI team. Not urgent but it was helpful for me to figure this stuff out, might be useful for other people to know how the CLI packages Python lambdas without having to do their own research.

**Review by edwardfoyle — APPROVED** [review]

**Comment by jazwiecki** [comment]
Figured out how to fix the spellcheck failure, sorry about that!

**Comment by ericclemmons** [comment]
Thanks for the updates! Merging this in...

**Pull request merged**
